pennylane/spin/spin_hamiltonian.py

```diff
@@ -689,3 +689,67 @@ def kitaev(n_cells, coupling=None, boundary_condition=False):
         hamiltonian += coeff * (opmap[op1](v1) @ opmap[op2](v2))
 
     return hamiltonian.simplify()
+
+
+def spin_hamiltonian(lattice):
+    r"""Generates a spin Hamiltonian for a custom lattice.
+
+    Args:
+        lattice (Lattice): custom lattice defined with custom_edges
+
+    Raises:
+        ValueError: if the provided Lattice does not have ``custom_edges`` defined with operators
+
+    Returns:
+        ~ops.op_math.Sum: Hamiltonian for the lattice
+
+    **Example**
+
+    .. code-block:: python
+
+        lattice = qml.spin.Lattice(
+            n_cells=[2, 2],
+            vectors=[[1, 0], [0, 1]],
+            positions=[[0, 0], [1, 5]],
+            boundary_condition=False,
+            custom_edges=[[(0, 1), ("XX", 0.5)], [(1, 2), ("YY", 0.6)], [(1, 4), ("ZZ", 0.7)]],
+            custom_nodes=[[0, ("X", 0.5)], [1, ("Y", 0.3)]],
+        )
+
+        >>> qml.spin.spin_hamiltonian(lattice=lattice)
+        >>> (
+                0.5 * (X(0) @ X(1))
+                + 0.5 * (X(2) @ X(3))
+                + 0.5 * (X(4) @ X(5))
+                + 0.5 * (X(6) @ X(7))
+                + 0.6 * (Y(1) @ Y(2))
+                + 0.6 * (Y(5) @ Y(6))
+                + 0.7 * (Z(1) @ Z(4))
+                + 0.7 * (Z(3) @ Z(6))
+                + 0.5 * X(0)
+                + 0.3 * Y(1)
+            )
```

Comment:
Check this output once (the indentation of the last bracket).

Reply:
Yes, this needs to be updated. Please be consistent with [other functions](https://github.com/PennyLaneAI/pennylane/blob/79cb2eb02a1d95ec269160a502eddb5d036b5f4b/pennylane/spin/spin_hamiltonian.py#L65).

Reply:
Looks right to me when I compare it to what I've got on my notebook.